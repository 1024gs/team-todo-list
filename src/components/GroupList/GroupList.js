import React from 'react';
import './GroupList.scss';

const GroupList = ({items, onClick}) => {
  return (
    <div className="GroupList">
      <ul className="collection">
        {items.map((x, i) => (
          <li
            className={`collection-item${x.active ? ' active' : ''}`}
            key={i}
            onClick={() => onClick(x)}
          >
            <i className="material-icons left">list</i>
            {x.title}
          </li>
        ))}
      </ul>
    </div>
  );
};

export default GroupList;
