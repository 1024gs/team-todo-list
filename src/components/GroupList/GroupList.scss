.GroupList .collection-item {
  cursor: pointer;
}
