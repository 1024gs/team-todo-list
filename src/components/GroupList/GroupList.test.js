import React from 'react';
import GroupList from './GroupList';
import {shallow} from 'enzyme';

const setup = (propOverrides) => {
  const props = Object.assign(
    {items: [{title: 'Team To-do List', active: true}]},
    propOverrides
  );
  const wrapper = shallow(<GroupList {...props} />);

  return {
    props,
    wrapper
  };
};

describe('GroupList', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('has an active item', () => {
    const {wrapper} = setup({
      items: [{title: 'a'}, {title: 'b', active: true}]
    });
    expect(
      wrapper
        .find('.collection-item')
        .at(0)
        .hasClass('active')
    ).toBe(false);
    expect(
      wrapper
        .find('.collection-item')
        .at(1)
        .hasClass('active')
    ).toBe(true);
  });
});
