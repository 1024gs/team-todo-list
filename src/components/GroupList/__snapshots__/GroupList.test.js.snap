// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList renders 1`] = `
<div
  className="GroupList"
>
  <ul
    className="collection"
  >
    <li
      className="collection-item active"
      key="0"
    >
      <i
        className="material-icons left"
      >
        list
      </i>
      Team To-do List
    </li>
  </ul>
</div>
`;
